packages/gatsby-theme-docs/src/components/code-example.js

```diff
@@ -0,0 +1,45 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { CodeBlock, ContentNotifications } from '@commercetools-docs/ui-kit';
+import useCodeExamples from '../hooks/use-code-examples';
+
+function CodeExample(props) {
+  const codeExamples = useCodeExamples();
+
+  const codeExample = codeExamples.find(example => {
+    return example.absolutePath.includes(props.file);
+  });
+
+  if (!codeExample) {
+    return reportError('Code example does not exist');
+  }
+
+  const language = props.file.split('.').pop();
```

Comment:
You have put the extension property into your GraphQL surface so I think you should use it here, too instead of re-parsing it from the file path.

Reply:
Agree